### PR TITLE
Add tests and improve add_classifications 

### LINKED ISSUE #6: Fix error for empty latitude/longitude
When a user passes a CSV with empty values for latitude/longitude a 500 error is returned from the **GetCDLStat** API endpoint.

### LINKED ISSUE #9: Fix 500 Server error for some CSVs
For some CSVs the following error occurs:
```
500 Server Error: Internal Server Error for url: https://nassgeodata.gmu.edu/axis2/services/CDLService/GetCDLStat
```

### PR DESCRIPTION
**Pull request by johnbradley**
Adds tests for the main functions.
Adds setup to automatically run the tests when a commit is pushed.
Changes add_classifications to return the resulting dataframe instead of modifying the input dataframe.

This is in preparation for releasing the package to the [python package index](https://pypi.org/).

Fixes #6
Fixes #9

### CODE REVIEW

File: src/cover/classify.py [hunk 1]
```diff
@@ -31,14 +31,14 @@ def lookup_classification(name_lookup, crop_scape_value):
 
 
 def read_crop_scape_csv(path):
-    df = pandas.read_csv(path, engine='python', sep=", ")
+    df = pd.read_csv(path, engine='python', sep=", ")
```
Comment (thompsonmj) [inline src/cover/classify.py:34]: Might be better to remove the space from the delimiter for flexibility:
`df = pd.read_csv(path, engine='python', sep=",")`

File: src/cover/classify.py [hunk 2]
```diff
@@ -54,23 +54,39 @@ def get_classification_fraction(df):
...
+    """
+    Returns a new dataframe with *_big and *_small land coverage columns added to df.
+    Empty lat/lon columns will result in all zeros for the categories.
 
-    classification_ary = df.apply(classify_row, axis=1, column_name='bb_small')
+    :param df: dataframe with latitude and longitude columns
+    :param lat_col: name of the latitude column in df
+    :param lon_col: name of the longitude column in df
+    """
+    df_enh = df.copy(deep=True)
```
Comment (thompsonmj) [inline src/cover/classify.py:80]: I assume the data will not be large enough where memory becomes an issue?

File: src/cover/classify.py [hunk 3]
```diff
@@ -54,23 +54,39 @@ def get_classification_fraction(df):
...
+    classification_ary = df_enh.apply(classify_row, axis=1, column_name='bb_small')
     for classificationType in CLASSIFICATION_TYPES:
-        df[classificationType + "_small"] = [x.get(classificationType, 0.0) for x in classification_ary]
+        df_enh[classificationType + "_small"] = [x.get(classificationType, 0.0) for x in classification_ary]
 
-    classification_ary = df.apply(classify_row, axis=1, column_name='bb_big')
+    classification_ary = df_enh.apply(classify_row, axis=1, column_name='bb_big')
     for classificationType in CLASSIFICATION_TYPES:
-        df[classificationType + "_big"] = [x.get(classificationType, 0.0) for x in classification_ary]
+        df_enh[classificationType + "_big"] = [x.get(classificationType, 0.0) for x in classification_ary]
```
Comment (thompsonmj) [inline src/cover/classify.py:90]: Possibly separate code for `"_small"` and `"_big"` into a function rather than repeated `for` loops.

**Review by thompsonmj — APPROVED** [review]
Couple minor comments.

File: src/cover/classify.py (hunk 1 above)
Comment (johnbradley) [reply src/cover/classify.py:34]: The CSV returned by cropspace uses ", " as the delimiter for all rows. If we made this change we would have to strip whitespace on all column names and values.

**Pull request merged**